bin/propolis-server/src/lib/initializer.rs

```diff
@@ -610,74 +617,93 @@ impl<'a> MachineInitializer<'a> {
             let StorageBackendInstance { be: backend, crucible } = self
                 .create_storage_backend_from_spec(
                     &disk.backend_spec,
-                    backend_name,
+                    backend_id,
                     &nexus_client,
                 )
                 .await?;
 
-            self.block_backends.insert(backend_name.clone(), backend.clone());
+            self.block_backends.insert(backend_id.clone(), backend.clone());
             let block_dev: Arc<dyn block::Device> = match device_interface {
                 DeviceInterface::Virtio => {
                     let vioblk = virtio::PciVirtioBlock::new(0x100);
 
-                    self.devices
-                        .insert(format!("pci-virtio-{}", bdf), vioblk.clone());
+                    self.devices.insert(device_id.clone(), vioblk.clone());
                     block::attach(vioblk.clone(), backend).unwrap();
                     chipset.pci_attach(bdf, vioblk.clone());
                     vioblk
                 }
                 DeviceInterface::Nvme => {
+                    let spec::StorageDevice::Nvme(nvme_spec) =
+                        &disk.device_spec
+                    else {
+                        unreachable!("disk is known to be an NVMe disk");
+                    };
+
                     // Limit data transfers to 1MiB (2^8 * 4k) in size
                     let mdts = Some(8);
-                    let component = format!("nvme-{}", disk_name);
+                    let component = format!("nvme-{}", device_id);
                     let nvme = nvme::PciNvme::create(
-                        disk_name.to_owned(),
+                        &nvme_spec.serial_number,
```

Comment:
Does this mean the disk serial number is showing up somewhere?
Did this mean the disk name was also somewhere in the past?  
Would a change here effect existing instances that had booted with `disk_name` in the past?

Reply:
The guest can get at the serial number via NVMe controller commands. Linux guests will use it (for example) to name the disk's devfs nodes (`/dev/disk/by-id/foo` where `foo` is derived in part from the serial number).

On the Omicron side, whoever fills out this `nvme_spec` will populate the `serial_number` field using the corresponding Omicron disk name (with whatever padding regime is needed to maintain compatibility). https://github.com/oxidecomputer/propolis/pull/816#pullrequestreview-2455445757 has some more context here.

Reply:
Great, so as long as we don't mess up the Omicron side :)